plutus-pab/src/Cardano/Node/Mock.hs

```diff
@@ -51,22 +51,25 @@ consumeEventHistory stateVar =
 
 addTx ::
     ( Member (LogMsg PABServerLogMsg) effs
-    , Member (Reader Client.TxSendHandle) effs
+    , Member (Reader (Maybe Client.TxSendHandle)) effs
     , MonadIO m
     , LastMember m effs
     )
  => Tx -> Eff effs NoContent
 addTx tx = do
     logInfo $ BlockOperation $ NewTransaction tx
     clientHandler <- Eff.ask
-    liftIO $ Client.queueTx clientHandler tx
+    case clientHandler of
+      Nothing      -> logError TxSendCalledWithoutMock
```

Comment:
Throw error here?

Reply:
There is no error throwing effect on this function. I figured it's not really supposed to throw errors. Do you think we should change that?

Reply:
Also, would we make that an effect error, or a Control.Exception one?

Reply:
I will add an error effect and see what happens. I think it's reasonable to throw an error if this api is used when it shouldn't be.

Reply:
Meta idea: Maybe we should make this distinction at type level so we cannot write code that triggers this behaviour (but this would be another PR).

Reply:
Yeah makes sense. Create a ticket for that or add a TODO comment so we don't forget?

Reply:
This code is only used in the mocked server so I think logging an error is good enough since this is only used for testing.

Reply:
TODO: https://github.com/input-output-hk/plutus-apps/issues/264